### DISCUSSION TITLE
Is it possible to set an error for a specific entity in the store?

### DISCUSSION THREAD

**Discussion by mikesnare** [salesforce/akita#684] [Ideas]
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[ ] Bug report  
[ ] Performance issue
[ ] Feature request
[ ] Documentation issue or request
[x] Support request
[ ] Other... Please describe:
</code></pre>

## Current behavior

Is there a way to flag a specific entity as being in error vs the entire store being in error?

If you have a container component that, say, iterates a bunch of entities and creates a new component for each and each component allows inline editing of that entity, is there any way to distinguish errors for entity 1 from errors for entity 2, etc.?

Without that, I think that each component will need to subscribe to the update call AND add the appropriate pipes to the observable in order to catch errors specific to their item, instead of just using an async pipe for errors associated with their own item.

## Expected behavior

Something like:
```ts
// component file:
onUpdateClick() {
  // Note that this 
  this.service.doSomeThing(this.form.value).subscribe();
}
```

```ts
// service file:
public doSomeThing(item: Partial<Item>) {
  return this.apiService.post('whatever', item).pipe(
    tap(i => this.store.update(i.id, i)),
    catchError(err => {
      this.store.setError(item.id, err);
      throw err;
    })
  )
}
```

And then in the component for the individual entity:
```html
<div class="error" *ngIf="query.getError(entity.id) | async">{{ query.getError(entity.id) | async }}</div>
```

## What is the motivation / use case for changing the behavior?

To allow for simpler error handling with EntityStores and queries.

**Comment by NetanelBasal** [comment]
You can use a UI store or manage a map where the key is the entity id, and the value is the error.

**Comment by mikesnare** [comment]
Sure.  I think it would be a nice addition, fwiw.  Not something I have time to issue a PR, for, though.  Thanks for the response.
